tests/keras/layers/convolutional_test.py

```diff
@@ -166,23 +165,21 @@ def test_convolution_2d_channels_last():
 @pytest.mark.skipif((K.backend() == 'cntk'),
                     reason='cntk only supports dilated conv on GPU')
 def test_convolution_2d_dilation():
-    num_samples = 2
-    filters = 2
-    stack_size = 3
-    kernel_size = (3, 2)
-    num_row = 7
-    num_col = 6
-    padding = 'valid'
+    input_data = np.arange(126).reshape((1, 7, 6, 3)).astype(np.float32)
+    expected_output = np.float32(
+        [[[[720, 720], [774, 774], [828, 828], [882, 882]],
+          [[1044, 1044], [1098, 1098], [1152, 1152], [1206, 1206]],
+          [[1368, 1368], [1422, 1422], [1476, 1476], [1530, 1530]]]]
+    ).reshape((1, 3, 4, 2))
```

Comment:
It should be better to test the correctness of the result in the backends tests, by making sure that all backends output the same values. It has the advantage that we know exactly what backend operation doesn't work and we don't have to hard-code the result.

See : https://github.com/keras-team/keras/blob/06c3a804e7d56cc77f76a28ee34aa67a34daedee/tests/keras/backend/backend_test.py#L1081

Do you think you could do something like that?

Reply:
+1, a fix for a backend op should be tested at the level of backend unit tests (to keep unit tests small, modular, and focused).